### Step 1. 필요한 라이브러리 임포트

In [1]:
import pandas as pd
import numpy as np

### Step 2. [데이터](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv) 확인

- order_id 주문번호
- quantity 주문 개수
- item_name 주문한 상품명
- choice_description 선택사항
- item_price 가격

### Step 3. 데이터 불러오기


**read_csv를 통해 데이터를 올바른 형태로 불러온 뒤 chipo에 저장해주세요**

In [2]:
url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv'
    
chipo = pd.read_csv(url, sep='\t')

### Step 4. 초반 10행 조회

In [3]:
chipo.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

### Step 5. 데이터셋은 몇 개의 행으로 이루어져 있나요? (최소 2가지 이상 풀이법으로 풀어주세요)

4622

In [4]:
print(chipo.shape)

(4622, 5)


In [5]:
print(chipo.count)

<bound method DataFrame.count of       order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                       

### Step 6. 데이터셋은 몇 개의 열로 이루어져 있나요?

5

In [6]:
len(chipo.columns)

5

### Step 7. 열(column)의 이름들을 모두 조회해주세요

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [7]:
chipo.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

### Step 8. 데이터셋의 인덱스를 조회해주세요

RangeIndex(start=0, stop=4622, step=1)

In [8]:
chipo.index

RangeIndex(start=0, stop=4622, step=1)

### Step 9. 가장 많은 개수가 주문된 item_name은 무엇인가요?

Chicken Bowl	

In [9]:
t1 = chipo.groupby('item_name').sum()['order_id']
t1[t1==t1.max()].idxmax()

'Chicken Bowl'

### Step 10. 가장 많이 주문된 item은 몇 개 주문되었나요?

761

In [10]:
c = chipo.groupby('item_name')
c = c.sum()
c = c.sort_values(['quantity'], ascending=False)
price = int(c.head(1)['quantity'].describe()['max'])
print(price)

761


### Step 11. choice_description에서 가장 많이 주문된건 어떤 항목인가요?

Diet Coke 159

In [11]:
t1 = chipo['choice_description'].value_counts()

In [12]:
t1 = chipo['choice_description'].value_counts()
t2 = t1[t1 == t1.max()].reset_index()
t2

choice_description  count
0        [Diet Coke]    134

### Step 12. 총 몇 개의 item이 주문되었나요? (quantity의 총 합을 구해주세요)

4972

In [13]:
print(chipo['quantity'].sum())

4972


### Step 13. item price를 float로 변경해주세요

#### Step 13.a. item_price의 데이터타입을 조회해주세요

dtype('O')

In [14]:
chipo.dtypes['item_name']

dtype('O')

#### Step 13.b. item_price에서 $기호를 제거하고 숫자 부분을 float으로 형변환 후 기존 dataframe에 저장해주세요

람다함수를 활용하면 편리합니다

In [15]:
chipo['item_price'] = chipo['item_price'].map(lambda x : float(x.replace('$','')))
chipo

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  
0                                                   NaN        2.39  
1                                          [Clementine]        3.39  
2                                               [Apple]        3.39  
3                                                   NaN        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
...                                                 ...         ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75  

[4622 rows x 5 columns]

#### Step 13.c. 변경된 item_price의 데이터타입을 조회해주세요

dtype('float64')

In [16]:
chipo.dtypes['item_price']

dtype('float64')

### Step 14. 총 매출은 얼마인가요?

39237.02

In [17]:
print((chipo['quantity'] * chipo['item_price']).sum())

39237.02


### Step 15. 총 몇 건의 주문이 발생헀나요? (주문번호 기준) 

1834

**4가지 해답을 적어뒀습니다. 이 해답을 참조하여 Step 17을 해결해주세요**

In [18]:
# Solution 1
print(chipo['order_id'].max())

1834


### Step 16. 주문당 평균 매출은 얼마인가요?

21.394231188658654

In [19]:
# Solution 1
총매출 = (chipo['quantity'] * chipo['item_price']).sum()
매출건수 = chipo['order_id'].max()
print(총매출/매출건수)


21.39423118865867


In [20]:
# Solution 2



### Step 17. 몇 종류의 item이 판매되었나요?

50

In [21]:
print(chipo.groupby('item_name').size().count())

50
